src/gasman.c

```diff
@@ -958,6 +958,14 @@ Bag * GlobalByCookie(
 
 static Bag NextMptrRestoring;
 
+#ifdef GAP_KERNEL_DEBUG
+BOOL InWorkspaceRestore(void)
+{
+    // Double negation turns pointer into a boolean
+    return !!NextMptrRestoring;
```

Comment:
or instead of being clever, we could just do an old-fashioned null pointer test? :-)
```suggestion
    return NextMptrRestoring != 0;
```